### ISSUE TITLE
varnish fails to pass its own testsuite

### ISSUE THREAD

**Issue by mjpieters** [Homebrew/homebrew-core#86516]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
% brew config
HOMEBREW_VERSION: 3.2.14-85-ged1d34e
ORIGIN: https://github.com/Homebrew/brew
HEAD: ed1d34e3e1e51160421d31165e71f23a0a75ff12
Last commit: 3 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 234115248fc858f361cfae575bcad28f22a4d23e
Core tap last commit: 46 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.Cnzew5g1tW/org.xquartz:0
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 16-core 64-bit kabylake
Clang: 13.0.0 build 1300
Git: 2.33.0 => /usr/local/bin/git
Curl: 7.64.1 => /usr/local/Homebrew/Library/Homebrew/shims/shared/curl
macOS: 11.6-x86_64
CLT: 13.0.0.0.1.1630607135
Xcode: 13.0
XQuartz: 2.8.1 => /opt/X11
% brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /Users/mj/.local/bin/pre-commit-validate-config
  /Users/mj/.pyenv/shims/python2-config
  /Users/mj/.pyenv/shims/python3.8d-config
  /Users/mj/.pyenv/shims/python3.6m-config
  /Users/mj/.pyenv/shims/python3.7-config
  /Users/mj/.pyenv/shims/python2.7-config
  /Users/mj/.pyenv/shims/python3.10-config
  /Users/mj/.pyenv/shims/python3.7m-config
  /Users/mj/.pyenv/shims/python3.9-config
  /Users/mj/.pyenv/shims/python-config
  /Users/mj/.pyenv/shims/python3-config
  /Users/mj/.pyenv/shims/python3.6-config
  /Users/mj/.pyenv/shims/python3.8-config
```

### 

- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

varnish, as built by homebrew, fails its own test suite. When built outside of the brew build environment, the testsuite passes.

I had noticed two issues that prompted me to try this out:

- varnish 7.0.0 was released in a form that [won't normally compile on macos](https://github.com/varnishcache/varnish-cache/issues/3699)
- a varnish extension I package has a test suite that runs into the same issue, but the problem didn't stem in that package as it wasn't reproducible with a locally built varnishd binary.

### What happened (include all command output)?

Varnish is tested using the `varnishtest` command; the following code extracts the test files from the source tarball and then runs those:

```sh
# fetch the source tarball
brew fetch -s varnish
# ask brew where it was downloaded to
tarball=$(brew --cache -s varnish)
# extract just the tests to a temporary directory
extract_to=$(mktemp -dt vtc)
pushd "${extract_to}"
tar xzf "${tarball}" --strip-components=4 varnish-7.0.0/bin/varnishtest/tests/
# run the tests (run in parallel based on the number of cores without overwhelming a machine)
# only report test failures, report all test failures (not just the first)
varnishtest -j $(($(sysctl -n hw.ncpu) / 2)) -qk *.vtc
# re-run the tests but only output the failure lines
varnishtest -j $(($(sysctl -n hw.ncpu) / 2)) -k *.vtc | grep '^----'
popd
```

This outputs a series of test failures:

<details>
  <summary>2x 165 test failure lines plus two status lines, click to expand</summary>

```
#    top  TEST b00000.vtc FAILED (0.868) exit=2
#    top  TEST b00001.vtc FAILED (0.971) exit=2
#    top  TEST b00002.vtc FAILED (1.165) exit=2
#    top  TEST b00003.vtc FAILED (1.173) exit=2
#    top  TEST b00012.vtc FAILED (1.175) exit=2
#    top  TEST b00013.vtc FAILED (1.271) exit=2
#    top  TEST b00015.vtc FAILED (1.387) exit=2
#    top  TEST b00017.vtc FAILED (1.070) exit=2
#    top  TEST b00020.vtc FAILED (3.363) exit=2
#    top  TEST b00018.vtc FAILED (5.219) exit=2
#    top  TEST b00037.vtc FAILED (0.867) exit=2
#    top  TEST b00023.vtc FAILED (6.156) exit=2
#    top  TEST b00038.vtc FAILED (1.069) exit=2
#    top  TEST b00053.vtc FAILED (0.859) exit=2
#    top  TEST b00055.vtc FAILED (1.280) exit=2
#    top  TEST b00056.vtc FAILED (0.970) exit=2
#    top  TEST b00052.vtc FAILED (2.939) exit=2
#    top  TEST b00066.vtc FAILED (1.180) exit=2
#    top  TEST b00064.vtc FAILED (1.698) exit=2
#    top  TEST b00073.vtc FAILED (1.062) exit=2
#    top  TEST c00000.vtc FAILED (1.073) exit=2
#    top  TEST c00002.vtc FAILED (0.965) exit=2
#    top  TEST c00014.vtc FAILED (1.495) exit=2
#    top  TEST c00019.vtc FAILED (0.966) exit=2
#    top  TEST c00013.vtc FAILED (1.889) exit=2
#    top  TEST c00023.vtc FAILED (1.060) exit=2
#    top  TEST c00020.vtc FAILED (1.590) exit=2
#    top  TEST c00034.vtc FAILED (0.970) exit=2
#    top  TEST c00036.vtc FAILED (1.178) exit=2
#    top  TEST c00044.vtc FAILED (0.859) exit=2
#    top  TEST c00045.vtc FAILED (0.861) exit=2
#    top  TEST c00046.vtc FAILED (0.859) exit=2
#    top  TEST c00049.vtc FAILED (1.897) exit=2
#    top  TEST c00050.vtc FAILED (2.933) exit=2
#    top  TEST c00061.vtc FAILED (0.863) exit=2
#    top  TEST c00058.vtc FAILED (1.796) exit=2
#    top  TEST c00070.vtc FAILED (1.060) exit=2
#    top  TEST c00071.vtc FAILED (1.380) exit=2
#    top  TEST c00072.vtc FAILED (1.274) exit=2
#    top  TEST c00073.vtc FAILED (1.167) exit=2
#    top  TEST c00075.vtc FAILED (1.066) exit=2
#    top  TEST c00089.vtc FAILED (0.962) exit=2
#    top  TEST c00081.vtc FAILED (3.984) exit=2
#    top  TEST c00080.vtc FAILED (5.112) exit=2
#    top  TEST c00091.vtc FAILED (0.964) exit=2
#    top  TEST c00063.vtc FAILED (9.800) exit=2
#    top  TEST c00096.vtc FAILED (0.971) exit=2
#    top  TEST c00097.vtc FAILED (2.217) exit=2
#    top  TEST c00098.vtc FAILED (2.004) exit=2
#    top  TEST c00099.vtc FAILED (1.907) exit=2
#    top  TEST c00102.vtc FAILED (1.697) exit=2
#    top  TEST d00007.vtc FAILED (1.180) exit=2
#    top  TEST d00008.vtc FAILED (1.378) exit=2
#    top  TEST d00009.vtc FAILED (1.179) exit=2
#    top  TEST d00010.vtc FAILED (1.283) exit=2
#    top  TEST d00012.vtc FAILED (1.174) exit=2
#    top  TEST d00011.vtc FAILED (2.527) exit=2
#    top  TEST d00032.vtc FAILED (1.901) exit=2
#    top  TEST d00033.vtc FAILED (2.099) exit=2
#    top  TEST d00034.vtc FAILED (1.996) exit=2
#    top  TEST d00035.vtc FAILED (1.692) exit=2
#    top  TEST d00013.vtc FAILED (2.725) exit=2
#    top  TEST d00037.vtc FAILED (1.383) exit=2
#    top  TEST e00000.vtc FAILED (1.383) exit=2
#    top  TEST e00001.vtc FAILED (1.374) exit=2
#    top  TEST e00002.vtc FAILED (1.587) exit=2
#    top  TEST e00003.vtc FAILED (1.589) exit=2
#    top  TEST d00036.vtc FAILED (2.837) exit=2
#    top  TEST e00004.vtc FAILED (1.384) exit=2
#    top  TEST d00038.vtc FAILED (2.937) exit=2
#    top  TEST e00005.vtc FAILED (1.280) exit=2
#    top  TEST e00006.vtc FAILED (1.384) exit=2
#    top  TEST e00007.vtc FAILED (1.273) exit=2
#    top  TEST e00008.vtc FAILED (1.070) exit=2
#    top  TEST e00009.vtc FAILED (1.182) exit=2
#    top  TEST e00010.vtc FAILED (1.284) exit=2
#    top  TEST e00011.vtc FAILED (1.383) exit=2
#    top  TEST e00012.vtc FAILED (1.383) exit=2
#    top  TEST e00013.vtc FAILED (1.385) exit=2
#    top  TEST e00014.vtc FAILED (1.281) exit=2
#    top  TEST e00015.vtc FAILED (1.379) exit=2
#    top  TEST e00016.vtc FAILED (1.273) exit=2
#    top  TEST e00017.vtc FAILED (1.490) exit=2
#    top  TEST e00020.vtc FAILED (1.280) exit=2
#    top  TEST e00021.vtc FAILED (1.282) exit=2
#    top  TEST e00019.vtc FAILED (1.689) exit=2
#    top  TEST e00022.vtc FAILED (1.373) exit=2
#    top  TEST e00023.vtc FAILED (1.276) exit=2
#    top  TEST e00024.vtc FAILED (1.384) exit=2
#    top  TEST e00026.vtc FAILED (1.281) exit=2
#    top  TEST e00030.vtc FAILED (1.282) exit=2
#    top  TEST g00001.vtc FAILED (1.692) exit=2
#    top  TEST g00002.vtc FAILED (1.891) exit=2
#    top  TEST g00003.vtc FAILED (1.384) exit=2
#    top  TEST l00002.vtc FAILED (0.862) exit=2
#    top  TEST l00003.vtc FAILED (0.966) exit=2
#    top  TEST l00004.vtc FAILED (0.859) exit=2
#    top  TEST m00000.vtc FAILED (0.972) exit=2
#    top  TEST m00021.vtc FAILED (1.792) exit=2
#    top  TEST m00054.vtc FAILED (1.585) exit=2
#    top  TEST o00005.vtc FAILED (2.306) exit=2
#    top  TEST r00425.vtc FAILED (0.972) exit=2
#    top  TEST r00873.vtc FAILED (0.962) exit=2
#    top  TEST r00917.vtc FAILED (0.867) exit=2
#    top  TEST r01030.vtc FAILED (0.962) exit=2
#    top  TEST r01038.vtc FAILED (1.265) exit=2
#    top  TEST r01092.vtc FAILED (0.961) exit=2
#    top  TEST r01140.vtc FAILED (0.966) exit=2
#    top  TEST r01175.vtc FAILED (0.862) exit=2
#    top  TEST r01284.vtc FAILED (1.385) exit=2
#    top  TEST r01468.vtc FAILED (1.894) exit=2
#    top  TEST r01499.vtc FAILED (4.407) exit=2
#    top  TEST r01608.vtc FAILED (1.090) exit=2
#    top  TEST r01624.vtc FAILED (1.382) exit=2
#    top  TEST r01730.vtc FAILED (0.967) exit=2
#    top  TEST r01739.vtc FAILED (1.082) exit=2
#    top  TEST r01746.vtc FAILED (0.972) exit=2
#    top  TEST r01648.vtc FAILED (3.974) exit=2
#    top  TEST r01755.vtc FAILED (0.970) exit=2
#    top  TEST r01490.vtc FAILED (11.839) exit=2
#    top  TEST r01764.vtc FAILED (0.967) exit=2
#    top  TEST r01821.vtc FAILED (1.179) exit=2
#    top  TEST r01914.vtc FAILED (0.970) exit=2
#    top  TEST r01857.vtc FAILED (2.204) exit=2
#    top  TEST r01990.vtc FAILED (0.968) exit=2
#    top  TEST r02233.vtc FAILED (1.896) exit=2
#    top  TEST r02270.vtc FAILED (0.975) exit=2
#    top  TEST r02266.vtc FAILED (1.593) exit=2
#    top  TEST r02305.vtc FAILED (1.178) exit=2
#    top  TEST r02310.vtc FAILED (0.956) exit=2
#    top  TEST r02258.vtc FAILED (3.020) exit=2
#    top  TEST r02351.vtc FAILED (0.959) exit=2
#    top  TEST r02406.vtc FAILED (1.290) exit=2
#    top  TEST r02372.vtc FAILED (2.015) exit=2
#    top  TEST r02527.vtc FAILED (0.859) exit=2
#    top  TEST r02530.vtc FAILED (0.867) exit=2
#    top  TEST r02589.vtc FAILED (1.076) exit=2
#    top  TEST r02831.vtc FAILED (1.077) exit=2
#    top  TEST r02964.vtc FAILED (0.850) exit=2
#    top  TEST r02946.vtc FAILED (4.062) exit=2
#    top  TEST r03006.vtc FAILED (2.941) exit=2
#    top  TEST r03266.vtc FAILED (0.866) exit=2
#    top  TEST r03038.vtc FAILED (5.524) exit=2
#    top  TEST r03354.vtc FAILED (0.966) exit=2
#    top  TEST t02001.vtc FAILED (0.963) exit=2
#    top  TEST t02002.vtc FAILED (0.862) exit=2
#    top  TEST t02003.vtc FAILED (0.961) exit=2
#    top  TEST s00007.vtc FAILED (4.900) exit=2
#    top  TEST t02000.vtc FAILED (5.927) exit=2
#    top  TEST t02013.vtc FAILED (1.891) exit=2
#    top  TEST t02011.vtc FAILED (3.046) exit=2
#    top  TEST t02022.vtc FAILED (1.487) exit=2
#    top  TEST v00003.vtc FAILED (0.967) exit=2
#    top  TEST v00004.vtc FAILED (1.613) exit=2
#    top  TEST v00012.vtc FAILED (1.365) exit=2
#    top  TEST v00010.vtc FAILED (2.870) exit=2
#    top  TEST v00042.vtc FAILED (1.783) exit=2
#    top  TEST v00043.vtc FAILED (1.592) exit=2
#    top  TEST v00006.vtc FAILED (12.274) exit=2
#    top  TEST v00055.vtc FAILED (1.481) exit=2
#    top  TEST v00051.vtc FAILED (2.005) exit=2
#    top  TEST v00059.vtc FAILED (1.274) exit=2
#    top  TEST v00062.vtc FAILED (1.485) exit=2
#    top  TEST v00058.vtc FAILED (3.456) exit=2
#    top  TEST v00064.vtc FAILED (1.583) exit=2
165 tests failed, 33 tests skipped, 597 tests passed
---- v1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c2    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c2    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- v1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c2    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c2    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c2    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- v1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c2    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- v1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- l2    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c2    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    HTTP rx EOF (fd:16 read: Undefined error: 0) 1
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c3    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- v1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c4    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c6    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c6    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- s2    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- v1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- v1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- v1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- v1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- v1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- v1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- v1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- v1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- v1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- v1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- v1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- v1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- v1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- v1    Bad exit status: 0x4000 exit 0x40 signal 0 core 0
---- c2    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c2    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- v1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c3    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- v1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- v1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c2    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c2    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- v1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c2    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- v1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- v1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- v1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c2    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c2    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- v1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- v1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- s1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
---- c1    Assert error in vsc_fill_point(), vsc.c line 240:  Condition(vt->type == VJSN_STRING) not true.  Errno=0 Undefined error: 0
165 tests failed, 33 tests skipped, 597 tests passed
```

</details>

If I instead compile the project _from an archive downloaded from GitHub_  (with a patch to fix , the test suite passes cleanly:

```
# assumption: all varnish build dependencies have been installed
# install additional autotools dependency
brew install automake autoconf autoconf-archive libtool
dlfile=$(mktemp -t varnish).tgz
curl -sL https://github.com/varnishcache/varnish-cache/archive/refs/tags/varnish-7.0.0.tar.gz -o ${dlfile}
builddir=$(mktemp -dt varnishbuild)
pushd ${builddir}
tar xzf ${dlfile} --strip-components=1
# fix for varnishcache/varnish-cache#3699, 7.0 fails to build on macos 10.15
curl -s https://github.com/varnishcache/varnish-cache/commit/20e007a5b17c1f68f70ab42080de384f9e192900.diff | patch -p1
./autogen.sh
./configure --disable-dependency-tracking --prefix=$(brew --prefix varnish) --localstatedir=$(brew --prefix)/var
make
make check  # runs the same tests
```

I note that I used the GitHub tarball here; the official download was created using an older autoconf toolchain, and it could be that that's the cause of this issue.

### What did you expect to happen?

For the tests to pass. The tests not passing indicates something went wrong building varnish.

### Step-by-step reproduction instructions (by running `brew` commands)

(already covered under "what happened")

**Comment by mjpieters** [comment]
I can reproduce the same problem on Linux, so this is an upstream problem. However, it can be solved by running `./autogen.sh` before running `./configure`.

This does mean the build dependencies have to be updated to include autotools:

```rb
  depends_on "autoconf" => :build
  depends_on "autoconf-archive" => :build
  depends_on "automake" => :build
  depends_on "libtool" => :build
```

**Comment by MikeMcQuaid** [comment]
> Varnish is tested using the `varnishtest` command; the following code extracts the test files from the source tarball and then runs those:

Is this something that is (or can be) built as part of the build process and installed? If so, we can (and should) add it to the `test do` block.

**Comment by mjpieters** [comment]
I

> Is this something that is (or can be) built as part of the build process and installed? If so, we can (and should) add it to the `test do` block.

Absolutely:

```ruby
    system "tar", "-xzf", cached_download, "-C", testpath, "--strip-components=4", "varnish-*/bin/varnishtest/tests/"
    # run the tests (run in parallel based on the number of cores without overwhelming a machine)
    # only report test failures, report all test failures (not just the first)
    parallel = Hardware::CPU.cores / 2
    system "#{bin}/varnishtest", "-j", parallel, "-qk", testpath/"*.vtc"
```

I'll make that a PR, together with the `./autogen.sh` fix; without running that the tests fail, with it, the tests pass.

**Comment by mjpieters** [comment]
Actually, the tests fail in the exact same way. I can only get them to pass when:

- I use `./autogen.sh`
- I *don't use the homebrew build system*, just download, patch, run autogen, configure and then make.

I don't know how to make the build work using the homebrew build system.

**Comment by mjpieters** [comment]
Good news: if I add the tests to my #88464 PR (which includes the `-flat_namespace` patch), then the tests pass, so this issue is due to misdetection of the MacOS version in libtool.

Should I add the tests to that PR or in a separate PR?

**Comment by carlocab** [comment]
Probably better to add it in a separate PR after the current one is merged to avoid having to wait for reviews on the test changes.

**Comment by mjpieters** [comment]
> Probably better to add it in a separate PR after the current one is merged to avoid having to wait for reviews on the test changes.

Check. I have it ready in the pipeline for when that one lands.
